pkg/controller/metrics/diverters/MetricsDispatcher.go

```diff
@@ -0,0 +1,55 @@
+package diverters
+
+import (
+	"fmt"
+	"github.com/opensds/opensds/pkg/model"
+)
+
+// A buffered channel that we can send work requests on.
+var MetricsQueue = make(chan model.MetricSpec, 100)
```

Comment:
I would suggest changing the element to `*model.MetricSpec`.

Reply:
updated.